### PR TITLE
Improve init command, use eslint:recommended ruleset

### PR DESCRIPTION
**Pull request by Dionysusnu**

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -58,9 +59,13 @@
 		"@typescript-eslint/explicit-module-boundary-types": "off",
 		"@typescript-eslint/no-require-imports": "error",
 		"@typescript-eslint/no-unused-expressions": "warn",
+		"curly": ["warn", "multi-line", "consistent"],
 		"prefer-const": "warn",
+		"no-console": "warn",
```
Comment (sasial-dev) [inline .eslintrc:64]: Why did you change it to warn from error?
Don't we *want* no console to get people to use `print`, `warn` & `error`?

Comment (Dionysusnu) [reply .eslintrc:64]: This is the compiler's eslint, not the user's. In the compiler, using console is no recommended, but not really an error, and it's annoying while debugging.

Comment (sasial-dev) [reply .eslintrc:64]: Oh, okay! Thanks for clarifying.

File: src/CLI/commands/init.ts [hunk 2]
```diff
@@ -222,28 +224,37 @@ async function init(argv: yargs.Arguments<InitOptions>, mode: InitMode) {
...
 			}
 		}
 
 		await cmd(`${selectedPackageManager.devInstall} ${devDependencies.join(" ")}`);
+	});
 
-		// create .eslintrc.json
-		if (eslint) {
+	if (eslint) {
+		await benchmark("Configuring eslint..", async () => {
```
Comment (osyrisrblx) [inline src/CLI/commands/init.ts:257]:
```suggestion
		await benchmark("Configuring ESLint..", async () => {
```

File: src/CLI/commands/init.ts [hunk 3]
```diff
@@ -210,9 +211,10 @@ async function init(argv: yargs.Arguments<InitOptions>, mode: InitMode) {
 			pkgJson.scripts.prepublishOnly = selectedPackageManager.build;
 		}
 		await fs.outputFile(paths.packageJson, JSON.stringify(pkgJson, null, 2));
+	});
 
-		// git init
-		if (git) {
+	if (git) {
+		await benchmark("Initializing git..", async () => {
```
Comment (osyrisrblx) [inline src/CLI/commands/init.ts:217]:
```suggestion
		await benchmark("Initializing Git..", async () => {
```

File: src/CLI/commands/init.ts [hunk 4]
```diff
@@ -191,8 +191,9 @@ async function init(argv: yargs.Arguments<InitOptions>, mode: InitMode) {
 		},
 	]);
 
-	await benchmark("Initializing..", async () => {
-		const selectedPackageManager = packageManagerCommands[packageManager];
+	const selectedPackageManager = packageManagerCommands[packageManager];
+
+	await benchmark("Initializing package..", async () => {
```
Comment (osyrisrblx) [inline src/CLI/commands/init.ts:196]:
```suggestion
	await benchmark("Initializing project..", async () => {
```

File: src/Shared/classes/RojoResolver.ts [hunk 5]
```diff
@@ -173,11 +173,12 @@ export class RojoResolver {
 			let configJson: unknown;
 			try {
 				configJson = JSON.parse(fs.readFileSync(realPath).toString());
-			} catch (e) {}
-			if (isValidRojoConfig(configJson)) {
-				this.parseTree(path.dirname(rojoConfigFilePath), configJson.name, configJson.tree, doNotPush);
-			} else {
-				warn(`RojoResolver: Invalid configuration! ${ajv.errorsText(validateRojo.get().errors)}`);
+			} finally {
```
Comment (osyrisrblx) [inline src/Shared/classes/RojoResolver.ts:176]: Could you explain this change?

File: src/LuauAST/util/isValidNumberLiteral.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-const DECIMAL_LITERAL_REGEX = /^(?:\d[\d_]*(?:\.[\d_]*)?|\.\d[\d_]*)(?:[eE][\+\-]?_*\d[\d_]*)?$/;
+const DECIMAL_LITERAL_REGEX = /^(?:\d[\d_]*(?:\.[\d_]*)?|\.\d[\d_]*)(?:[eE][+-]?_*\d[\d_]*)?$/;
```
Comment (osyrisrblx) [inline src/LuauAST/util/isValidNumberLiteral.ts:1]: Did ESLint recommend this change?

**Review by osyrisrblx — CHANGES_REQUESTED** [review]

File: src/LuauAST/util/isValidNumberLiteral.ts (hunk 6 above)
Comment (Dionysusnu) [reply src/LuauAST/util/isValidNumberLiteral.ts:1]: Yes. + and - don't need to be escaped inside character sets, because they are always literal in there.

File: src/CLI/commands/init.ts (hunk 4 above)
Comment (Dionysusnu) [reply src/CLI/commands/init.ts:196]: This isn't the project, but the npm package. However I do agree the current wording may be confusing for game projects. Any suggestions?

File: src/Shared/classes/RojoResolver.ts (hunk 5 above)
Comment (Dionysusnu) [reply src/Shared/classes/RojoResolver.ts:176]: This was part due to eslint not allowing an empty catch block. This made me rewrite it to look nicer. This way, it's obvious that the `isValidRojoConfig` check is related to the `configJson` parsing.

File: src/CLI/commands/init.ts (hunk 4 above)
Comment (osyrisrblx) [reply src/CLI/commands/init.ts:196]: `Initializing package.json..` maybe?

Comment (Dionysusnu) [reply src/CLI/commands/init.ts:196]: What about `Initializing npm package..`?

**Review by osyrisrblx — APPROVED** [review]
